### PR TITLE
Feature - Categories in Habits microservice

### PR DESCRIPTION
**Pull request by IGORnvk**
This branch introduces a functionality of categories and habit records.

In order to start testing, you should delete your `db.sqlite` file in habits microservice and run `npm install` in the frontend.
For testing:
- [x] Create a category
- [x] Edit a category
- [x] Delete a category
- [x] Add a habit to a category
- [x] Edit a habit of some category
- [x] Delete a habit of some category
- [x] Complete a habit by pressing "Done" in habit view
- [x] Check that checkmark turned purple when habit is done on the habits overview page

### CODE REVIEW

**Comment by v-stamenova** [comment]
In review

File: frontend/src/lib/components/CategoryItem.svelte [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    // import CircularProgressBar from './CircularProgressBar.svelte';
+    export let title;
+    export let categoryId;
+
+    const progress = 1;
+</script>
+
+<div class="bg-white rounded-xl p-1 flex justify-between" style="box-shadow: rgba(0, 0, 0, 0.15) 2.4px 2.4px 3.2px">
+    <a href={`categories/${categoryId}/habits`} class="flex w-full">
+        <img alt="routine icon" src="../../src/img/routineIcon{categoryId}.png" class="object-scale-down">
```
Comment (v-stamenova) [inline frontend/src/lib/components/CategoryItem.svelte:11]: When you add new categories their id increases and there are only 3 images, therefore no image is available and 404 gets thrown in the console.

File: frontend/src/routes/categories/create/+page.svelte [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                  {/each}
+                </ul>
+              </div>
+            {/if}
+
+            <AuthInput
+              name={"name"}
+              label={"Name"}
+              type={"text"}
+              placeholder={"School Rutine"}
```
Comment (v-stamenova) [inline frontend/src/routes/categories/create/+page.svelte:46]: Spelling mistake

File: frontend/src/routes/categories/[categoryId]/+page.svelte [hunk 3]
```diff
@@ -7,7 +7,7 @@
   let countdown;
 </script>
 
-<Header title="Habits" route="/habits" displayMenu=1 displayBackButton=1 />
+<Header title="No" route="/categories" displayMenu=1 displayBackButton=1 />
```
Comment (v-stamenova) [inline frontend/src/routes/categories/[categoryId]/+page.svelte:10]: What do you mean title "No"?

File: frontend/src/routes/categories/[categoryId]/+page.svelte
Comment (v-stamenova) [inline frontend/src/routes/categories/[categoryId]/+page.svelte:1]: Can't access that page - 500.

If you don't want it to be accessible, just don't create it.

**Review by v-stamenova — COMMENTED** [review]
Looks good (not a thorough review - no time), just figure out what you will be doing with those minor stuff I mentioned

File: frontend/src/routes/categories/[categoryId]/+page.svelte (hunk 3 above)
Comment (anafaraco01) [reply frontend/src/routes/categories/[categoryId]/+page.svelte:10]: Yes this page should not exist

File: frontend/src/routes/categories/[categoryId]/+page.svelte
Comment (anafaraco01) [reply frontend/src/routes/categories/[categoryId]/+page.svelte:1]: Page deleted

File: frontend/src/routes/categories/create/+page.svelte (hunk 2 above)
Comment (anafaraco01) [reply frontend/src/routes/categories/create/+page.svelte:46]: done

File: frontend/src/lib/components/CategoryItem.svelte (hunk 1 above)
Comment (anafaraco01) [reply frontend/src/lib/components/CategoryItem.svelte:11]: they will repeat for now

**Review by v-stamenova — APPROVED** [review]

**Pull request merged**